src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMyBatisRepository.java

```diff
@@ -78,10 +80,33 @@ public List<ClinicalDataCount> getClinicalDataCounts(StudyViewFilter studyViewFi
     }
 
     @Override
-    public List<GenomicDataCount> getGenomicDataCounts(StudyViewFilter studyViewFilter) {
+    public List<GenomicDataCount> getMolecularProfileSampleCounts(StudyViewFilter studyViewFilter) {
         CategorizedClinicalDataCountFilter categorizedClinicalDataCountFilter = extractClinicalDataCountFilters(studyViewFilter);
-        return mapper.getGenomicDataCounts(studyViewFilter, categorizedClinicalDataCountFilter,
+        var counts = mapper.getMolecularProfileSampleCounts(studyViewFilter, categorizedClinicalDataCountFilter,
             shouldApplyPatientIdFilters(studyViewFilter,categorizedClinicalDataCountFilter));
+
+        Map<String, List<GenomicDataCount>> postsPerType = counts.stream()
+            .collect((Collectors.groupingBy(GenomicDataCount::getValue)));
+
+        // different cancer studies combined into one cohort will have separate molecular profiles
+        // of a given type (e.g. mutation).  We need to merge the counts for these
+        // different profiles based on the type and choose a label
+        // this code just picks the first label, which assumes that the labels will match
+        // across studies. 
+        List<GenomicDataCount> mergedOutput = new ArrayList<>();
+        for (Map.Entry<String,List<GenomicDataCount>> entry : postsPerType.entrySet()) {
+            var dc = new GenomicDataCount();
+            dc.setValue(entry.getKey());
+            // here just snatch the label of the first profile
+            dc.setLabel(entry.getValue().get(0).getLabel());
+            Integer sum = entry.getValue().stream()
```

Comment:
I would just suggest updating the variable names.

mergedOutput, counts, postsPerType

Reply:
how's this